source/client/factories_impl.cc

```diff
@@ -94,15 +82,16 @@ OutputCollectorFactoryImpl::OutputCollectorFactoryImpl(Envoy::TimeSource& time_s
     : OptionBasedFactoryImpl(options), time_source_(time_source) {}
 
 OutputCollectorPtr OutputCollectorFactoryImpl::create() const {
-  const std::string format = options_.outputFormat();
-  if (format == "human") {
+  switch (options_.outputFormat()) {
+  case nighthawk::client::OutputFormat::HUMAN:
```

Comment:
Do you handle the DEFAULT cases here and elsewhere?

Reply:
`OptionImpl` will never return the `DEFAULT` enum default values here, hence the `NOT_REACHED_GCOVR_EXCL_LINE` below.  

Context: When we construct `OptionImpl`  from a `proto`, it uses `PROTOBUF_GET_WRAPPED_OR_DEFAULT` to either get the set value, or take the default value from the private field declared in the header (for POD types) when it is not set. [1] `PROTOBUF_GET_WRAPPED_OR_DEFAULT` considers the `DEFAULT` flag as not set as it doesn't get serialized.

[1] e.g. https://github.com/envoyproxy/nighthawk/pull/98/files/7d8ff35fcc49cb39e10e6acfefc5ecf2de5b48b5#diff-54af86e901ab4e3f97d4dcf19e326cc3R273

Reply:
Got it, makes sense.